docs/devel/hello-world-gadget.md

```diff
@@ -274,6 +283,11 @@ other things, it can be used to specify the format to be used.
 
 An initial version of the metadata file can be created by passing `--update-metadata` to the build command:
 
+> [!NOTE]
+> This is only valid if the eBPF program used one of the available macros to define the type of
+> the gadget (e.g. `GADGET_TRACER`). Check out the
+> [Implementing the eBPF program](#implementing-the-ebpf-program) section for more information.
+
```

Comment:
I'm not sure how to write it, but I'd like to avoid mentioning "type of the gadget" as this concept won't exist anymore with image-based gadgets.

Perhaps something like works?

```suggestion
> [!NOTE]
> This is only valid if the eBPF program used one the `GADGET_TRACER` macro to define a tracer. If such macro wasn't used, the metadata file has to be created by hand.

```

Reply:
Understood. I used:

```md
> [!NOTE]
> The `tracers` and `structs` sections will only be generated if the eBPF program defined a tracer
> using the `GADGET_TRACER` macro.
```